### PR TITLE
DeviceId and MACAddress filters.

### PR DESCRIPTION
**Pull request by mjleesment**
# Pull Request Title

> Please review our [Code of Conduct](https://github.com/CrowdStrike/falconpy/CODE_OF_CONDUCT.md) 
> and our [Contribution Guidelines](https://github.com/CrowdStrike/falconpy/CONTRIBUTING.md) before submitting a Pull Request.
>
> REMOVE ALL PULL REQUEST HINTS BEFORE SUBMITTING
>
> Pull Request general description should go here.
> Please fill out all values and then remove any help text before submitting your PR.
> Refer to [this PR](https://github.com/CrowdStrike/falconpy/pull/67) as a reference example!

- [ ] Enhancement
- [ ] Major feature update
- [ ] Bug fixes
- [ ] Breaking change
- [ ] Updated unit tests
- [ ] Documentation

> Check the values above that match your PR and remove the remaining.

## Bandit analysis

```shell
BANDIT ANALYSIS RESULTS SHOULD BE POSTED
HERE. PLEASE REVIEW CONTRIBUTING.md FOR
MORE DETAILS ON GENERATING BANDIT RESULTS
```

## Added features and functionality

If your PR adds features or functionality, what should be included in the next release notes?

## Issues resolved

- Bug fix: Please list related bugs individually. Identifying the ticket or bug report in the PR description auto-updates the affected ticket and helps the community with ticket triage. For example:
- Fixes https://github.com/CrowdStrike/falconpy/issues/1234 by doing foo 
- Mitigates https://github.com/CrowdStrike/falconpy/issues/5678 by updating bar.
- BE EXPLICIT in what you are resolving

## Other

- List any other details here
- Documentation regarding your changes can also be listed here

### CODE REVIEW

**Review by ChristopherHammond13 — APPROVED** [review]
Thank you for this! I can definitely get this merged in. However, I also want to draw your attention to #113 in which this filtering system is going to be replaced. I will add your filters to the `caracara-filters` project.

File: caracara/modules/hosts/hosts_filters.py [hunk 1]
```diff
@@ -268,6 +268,34 @@ def example(self) -> str:
...
+class HostDeviceIdFilterAttribute(FalconFilterAttribute):
+    """Filter by host's sensor id."""
+
+    name = "DeviceId"
+    fql = "device_id"
+
+    def example(self) -> str:
+        """Show filter example."""
+        return (
+            "This filter accepts a MAC address string associated with a "
```
Comment (ChristopherHammond13) [inline caracara/modules/hosts/hosts_filters.py:294]: The help text here isn't quite right. I'll get this rewritten for `caracara-filter`.

**Review by ChristopherHammond13 — CHANGES_REQUESTED** [review]
The Device ID filter isn't quite right, but I'll rewrite that in `caracara-filters` anyway :)

File: caracara/modules/hosts/hosts_filters.py (hunk 1 above)
Comment (mjleesment) [reply caracara/modules/hosts/hosts_filters.py]: Thanks, updated.

**Comment by ChristopherHammond13** [comment]
Given the new filter system, this PR has now been deprecated and the functionality has been implemented here: https://github.com/CrowdStrike/caracara-filters/issues/5

Thank you for your contribution and suggestion here! Please feel free to open PRs with more filters over at the caracara-filters repository.

**Pull request closed without merging**